src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellItemRenderer.cs

```diff
@@ -433,6 +435,18 @@ public override void ViewWillLayoutSubviews()
 			base.ViewWillLayoutSubviews();
 		}
 
+		async Task UpdateNavBarHidden(ShellSection shellSection)
+		{
+			if (SelectedViewController is UINavigationController navigationController && _displayedPage is not null)
+			{
+				// We need to wait for the layout to finish before we can set the navigation bar hidden,
+				// otherwise it will not animate correctly.
+				await Task.Yield();
```

Comment:
I think the `await Task.Yield();` approach is not totally reliable for waiting for layout to finish. It only yields the current thread back to the event loop for one cycle, which doesn't guarantee that layout operations will complete.

Reply:
Have some ideas (not tested in your code):
- Can use the `ViewDidLayoutSubviews` method ensuring it runs after layout updates. Actually, we invoke `_appearanceTracker?.UpdateLayout(this); `in this way.
- Try to use the UIThread using `UIApplication.SharedApplication.InvokeOnMainThread`.

Reply:
I believe that `ViewDidLayoutSubviews` is safest from the given options